examples/05-nautobot-peeringdb/adapter_nautobot.py

```diff
@@ -0,0 +1,201 @@
+                "longitude": attrs["longitude"],
+            },
+        )
+        return super().create(diffsync, ids=ids, attrs=attrs)
+
+    def update(self, attrs):
+        """Update an existing Site record in remote Nautobot.
+        Args:
+            attrs (dict): Updated values for this record's _attributes
+        """
+        data = {}
+        if "slug" in attrs:
+            data["slug"] = attrs["slug"]
+        if "description" in attrs:
+            data["description"] = attrs["description"]
+        if "status_slug" in attrs:
+            data["status"] = attrs["status_slug"]
+        if "region_name" in attrs:
+            if attrs["region_name"]:
+                data["region"] = {"name": attrs["region_name"]}
+            else:
+                data["region"] = None
+        if "latitude" in attrs:
+            data["latitude"] = attrs["latitude"]
+        if "longitude" in attrs:
+            data["longitude"] = attrs["longitude"]
+        self.diffsync.patch(f"/api/dcim/sites/{self.pk}/", data)
+        return super().update(attrs)
+
+    def delete(self):
+        """Delete an existing Site record from remote Nautobot."""
+        # self.diffsync.delete(f"/api/dcim/sites/{self.pk}/")
+        return super().delete()
+
+
+class NautobotRemote(DiffSync):
+    """DiffSync adapter class for loading data from a remote Nautobot instance using Python requests.
+    In a more realistic example, you'd probably use PyNautobot here instead of raw requests,
+    but we didn't want to add PyNautobot as a dependency of this plugin just to make an example more realistic.
+    """
+
+    # Model classes used by this adapter class
+    region = RegionNautobotModel
+    site = SiteNautobotModel
+
+    # Top-level class labels, i.e. those classes that are handled directly rather than as children of other models
+    top_level = ("region", "site")
+
+    def __init__(self, *args, url=NAUTOBOT_URL, token=NAUTOBOT_TOKEN, **kwargs):
+        """Instantiate this class, but do not load data immediately from the remote system.
+        Args:
+            url (str): URL of the remote Nautobot system
+            token (str): REST API authentication token
+            job (Job): The running Job instance that owns this DiffSync adapter instance
+        """
+        super().__init__(*args, **kwargs)
+        if not url or not token:
+            raise ValueError("Both url and token must be specified!")
+        self.url = url
+        self.token = token
+        self.headers = {
+            "Accept": "application/json",
+            "Authorization": f"Token {self.token}",
+        }
+
+    def load(self):
+        """Load Region and Site data from the remote Nautobot instance."""
+        # To keep the example simple, we disable REST API pagination of results.
+        # In a real job, especially when expecting a large number of results,
+        # we'd want to handle pagination, or use something like PyNautobot.
```

Comment:
Aren't you actually handling pagination with your `region_data["next"]` logic below?

Reply:
yes, not updated properly. you can guess where this copy&paste is coming from